lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -637,16 +665,26 @@ pub fn generate_create_account(
     space: proc_macro2::TokenStream,
     owner: proc_macro2::TokenStream,
     seeds_with_nonce: proc_macro2::TokenStream,
+    rent_exempt: &ConstraintRentExempt,
 ) -> proc_macro2::TokenStream {
-    quote! {
-        // If the account being initialized already has lamports, then
-        // return them all back to the payer so that the account has
-        // zero lamports when the system program's create instruction
-        // is eventually called.
+    let current_lamports = quote! {
         let __current_lamports = #field.to_account_info().lamports();
+    };
+
+    let lamports = match rent_exempt {
+        ConstraintRentExempt::Skip => quote! {0},
```

Comment:
I don't think this is necessarily what we want because it means the account will immediately be garbage collected.

Reply:
*at the end of the transaction so following ix can still use it.

I doubt this will be used much but to me it seems like the code a user would expect to be generated by that combination of constraints. I think the API is not great (not obvious that this causes init with 0) but at least it would provide the functionality

Reply:
but yea maybe its better to just roll out the new rent/lamports feature asap and forbid rent_exempt = skip on init in the parser

Reply:
Sounds good. `rent_exempt = skip` was originally added with the expectation that the account was created in a prior instruction (with what is now `#[account(zero)]`).

Reply:
just to confirm. The thing that sounds good to you is this https://github.com/project-serum/anchor/pull/1234#discussion_r779231266 comment, not the one before?